### ISSUE TITLE
Your MongoDB instance may be publicly accessible

### ISSUE THREAD

**Issue by fshmcallister** [WoTTsecurity/wott-io#208]
We detected that a MongoDB instance on {devices} may be accessible remotely. Consider either blocking port 27017 through the WoTT firewall management tool, or re-configure MongoDB to only listen on localhost. 

Part of #198

**Comment by Menmarin** [comment]
## Title
Remote access on MongoDB instance detected



## TL;DR
We detected that a MongoDB instance on {devices} may be accessible remotely. Consider either blocking port 27017 through the WoTT firewall management tool, or reconfigure MongoDB to only listen on localhost.


MongoDB is a NoSQL, document database that is widely used by companies and individuals. Its popularity and support makes it a contender for one of the best options for database needs. Unfortunately, its popularity makes it a target for experienced and mediocre attackers alike.

MongoDB comes with a plethora of features designed to ease the process of setting up and using the platform. With these features, comes a risk of exposing your instance to attacks. One such feature is remote access. The vulnerabilities it opens up by leaving it open can be devastating. If an attacker somehow was able to get their hands on login credentials from one of your employees, they could connect to your MongoDB instance through remote access and it wouldn't be flagged as they logged in with the correct credentials. In addition, traffic over MongoDB connections are not encrypted, therefore opening the avenue for an attacker to be listening/sniffing the data being transmitted from a remote host to the database.

Wott firewall management tool has a feature that allows you to block the port 27017, which is the port used by MongoDB for remote access. Additionally, you can reconfigure the MongoDB configuration file to only listen on localhost. To do this, navigate to the config file usually found in yout ```etc``` directory and add the following line:

```
bind_ip = 127.0.0.1
```

Please remember to restart your mongodb instance as this will not take effect without doing so. For more information on how to securely configure your MongoDB instance, WoTT has a [step-by-step guide](https://wott.io/blog/tutorials/2019/11/24/mongodb) to aid you.

**Comment by vpetersson** [comment]
@Menmarin This has the wrong format. Needs to be tl;dr *and* a longer version.

**Comment by fshmcallister** [comment]
From what I can see, I think the second paragraph onwards is the FAQ, he's just forgotten to add it in.

**Comment by fshmcallister** [comment]
**Suggested changes to FAQ**

MongoDB is a NoSQL document database that is widely used by companies and individuals. Its popularity and support makes it a contender for one of the best options for database needs. Unfortunately, its popularity makes it a target for experienced and mediocre attackers alike.

MongoDB contains a plethora of features designed to ease the process of setting up and using its platform. One such feature is remote access which can potentially expose your system to attack. Enabling remote access increases the risk of stolen credentials being used to access your MongoDB, as well as just generally exposing your data unnecessarily from outside your local network. Additionally, traffic over MongoDB connections are not encrypted, therefore opening an avenue for an attacker to be listening/sniffing the data being transmitted from a remote host to the database.

WoTT's firewall management tool has a feature that allows you to block the port 27017, which is the port used by MongoDB for remote access. Additionally, you can reconfigure the MongoDB configuration file to only listen on localhost. To do this, navigate to the config file usually found in yout `etc` directory and add the following line: 
```
bind_ip = 127.0.0.1
```
Please remember to restart your MongoDB instance as this change will not take effect until you do so. For more information on how to securely configure your MongoDB instance, WoTT has a step-by-step guide to aid you.

**Comment by vpetersson** [comment]
> traffic over MongoDB connections are not encrypted

*by default* - https://docs.mongodb.com/manual/tutorial/configure-ssl/

**Comment by fshmcallister** [comment]
updated

**Comment by vpetersson** [comment]
@fshmcallister This is missing the correct format as @a-martynovich correctly pointed out. Could you do this asap?

**Comment by fshmcallister** [comment]
**TITLE**
Remote access on MongoDB detected
> consider disabling remote access

**TL;DR**
We detected that a MongoDB instance may be accessible remotely. Consider either blocking port 27017 through the WoTT firewall management tool, or reconfigure MongoDB to only listen on localhost.


**FAQ**

MongoDB is a NoSQL document database that is widely used by companies and individuals. Its popularity and support makes it an ideal database option for developers. Unfortunately, its popularity makes it a target for experienced and amateur attackers alike.

MongoDB contains a plethora of features designed to ease the process of setting up and using its platform. One such feature is remote access which can potentially expose your system to attack. 
Enabling remote access increases the risk of stolen credentials being used to access your MongoDB, as well as leaving you unnecessarily vulnerable to attack by human error, such as unprotected remote login. Additionally, traffic over MongoDB connections is not encrypted by default, therefore opening an avenue for an attacker to be listening/sniffing the data being transmitted from a remote host to the database.

WoTT's firewall management tool has a feature that allows you to block the port 27017, which is the port used by MongoDB for remote access. Additionally, you can reconfigure the MongoDB configuration file to only listen on localhost. To do this, navigate to the config file, usually \`/etc/mongod.conf\` and add the following line: 
\`\`\`
bind_ip = 127.0.0.1
\`\`\`
Please remember to restart your MongoDB instance as this change will not take effect until you do so. For more information on how to securely configure your MongoDB instance, WoTT has a step-by-step guide to aid you.
